autoload/ale/fixers/ptop.vim

```diff
@@ -0,0 +1,17 @@
+" Author: BarrOff https://github.com/BarrOff
+" Description: Integration of ptop with ALE.
+
+call ale#Set('pascal_ptop_executable', 'ptop')
+call ale#Set('pascal_ptop_options', '')
+
+function! ale#fixers#ptop#Fix(buffer) abort
+    let l:executable = ale#Var(a:buffer, 'pascal_ptop_executable')
+    let l:options = ale#Var(a:buffer, 'pascal_ptop_options')
+
+    return {
+    \   'command': ale#Escape(l:executable)
+    \       . (empty(l:options) ? '' : ' ' . l:options)
+    \       . ' %t %t',
```

Comment:
ALE replaces all instances of %t with the same temporary file. This will result in ptop reading from and writing to the exact same file. This may result in corrupted output on some filesystems.

I think replacing this to ' %s %t' has the desired effect and would be safer.